stan/math/rev/core/var.hpp

```diff
@@ -63,185 +74,63 @@ class var {
-   * by constructing a new <code>vari</code> with the argument
-   * cast to <code>double</code>, and a zero adjoint.
-   *
-   * @param x Value of the variable.
-   */
-  // NOLINTNEXTLINE
-  var(unsigned short x) : vi_(new vari(static_cast<double>(x), false)) {}
-
-  /**
-   * Construct a variable from the specified arithmetic argument
-   * by constructing a new <code>vari</code> with the argument
-   * cast to <code>double</code>, and a zero adjoint.
-   *
-   * @param x Value of the variable.
-   */
-  // NOLINTNEXTLINE
-  var(unsigned int x)
-      : vi_(new vari(static_cast<double>(x), false)) {}  // NOLINT
+  var_value() : vi_(static_cast<vari_type*>(nullptr)) {}
 
   /**
-   * Construct a variable from the specified arithmetic argument
-   * by constructing a new <code>vari</code> with the argument
-   * cast to <code>double</code>, and a zero adjoint.
-   *
+   * Construct a variable from the specified arithmetic type argument
+   * by constructing a new `vari_value<value_type>`. For integral types the
+   * `vari_value<value_type>` will hold doubles. This constructor is only valid
+   * when `S` is convertible to this `vari_value`'s `value_type`.
+   * @tparam S A type that is convertible to `value_type`.
    * @param x Value of the variable.
    */
-  // NOLINTNEXTLINE
-  var(unsigned long x)
-      : vi_(new vari(static_cast<double>(x), false)) {}  // NOLINT
-
-#ifdef _WIN64
-  // these two ctors are for Win64 to enable 64-bit signed
-  // and unsigned integers, because long and unsigned long
-  // are still 32-bit
+  template <typename S, require_convertible_t<S&, value_type>* = nullptr>
+  var_value(const S& x) : vi_(new vari_type(x, false)) {}  // NOLINT
 
   /**
-   * Construct a variable from the specified arithmetic argument
-   * by constructing a new <code>vari</code> with the argument
-   * cast to <code>double</code>, and a zero adjoint.
-   *
-   * @param x Value of the variable.
+   * Construct a variable from a pointer to a variable implementation.
+   * @tparam S The type in the vari_value pointer that has the same `value_type`
+   *  as this `var_value`. For integral types a `vari_value<Integral>` is the
+   * same as `vari_value<double>` so can be `reinterpret_cast` without a copy.
+   * @param vi A vari_value pointer.
    */
-  var(size_t x) : vi_(new vari(static_cast<double>(x), false)) {}  // NOLINT
+  var_value(vari_value<T>* vi)  // NOLINT
+      : vi_(vi) {}
 
   /**
-   * Construct a variable from the specified arithmetic argument
-   * by constructing a new <code>vari</code> with the argument
-   * cast to <code>double</code>, and a zero adjoint.
-   *
-   * @param x Value of the variable.
+   * Constructor from `var_value` whose value_type is the same as this class's
+   * `value_type`. This is used in cases such as
+   * `var_value<double> a(4.0); var_value<int> b(a)` since the `value_type` for
+   *  a `var_value` with an integral type is a double.
```

Comment:
Another comment that is not up to date. This is just a copy constructor. Actually, do we need to explicitely define it? Even if we do you can just use: ` = default`.

Reply:
No we do not!